tests/phpunit/locallib_test.php

```diff
@@ -111,6 +111,8 @@ public function test_should_login_redirect($duallogin, $param, $session, $expect
                 $_GET['SimpleSAML_Auth_State_exceptionId'] = '...';
             } else if ($param == 'post') {
                 $_SERVER['REQUEST_METHOD'] = 'POST';
+                $_POST['username'] = 'testuser';
```

Comment:
test_should_login_redirect gets call many times with different data, this is what the @dataProvider should_login_redirect_testcases is for. 

So instead of hard coding this, make $post a new function argument and then pass it in as a new test scenario in the data provider and default it to null in all the existing dataproviders

Admittedly this test function has way too much logic in it and needs a refactor, it should do almost nothing other that creating the preconditions for should_login_redirect

Reply:
just a heads up this existing dodgy unit test will get fixed in another PR which is being worked on now, eta a day or so, will make this a bit simpler